ksql-execution/src/main/java/io/confluent/ksql/execution/function/UdafUtil.java

```diff
@@ -44,35 +45,58 @@ private UdafUtil() {
     try {
       final ExpressionTypeManager expressionTypeManager =
           new ExpressionTypeManager(schema, functionRegistry);
-      final List<Expression> functionArgs = functionCall.getArguments();
-      final Schema argumentType = expressionTypeManager.getExpressionSchema(functionArgs.get(0));
+
+      final Schema argumentType =
+          expressionTypeManager.getExpressionSchema(functionCall.getArguments().get(0));
 
       // UDAFs only support one non-constant argument, and that argument must be a column reference
-      final Expression arg = functionArgs.get(0);
+      final Expression arg = functionCall.getArguments().get(0);
       final OptionalInt udafIndex;
       if (arg instanceof ColumnReferenceExp) {
         udafIndex = schema.valueColumnIndex(((ColumnReferenceExp) arg).getReference());
       } else {
         // assume that it is a column reference with no alias
         udafIndex = schema.valueColumnIndex(ColumnRef.withoutSource(ColumnName.of(arg.toString())));
       }
+      if (!udafIndex.isPresent()) {
+        throw new KsqlException("Could not find column for expression: " + arg);
+      }
 
-      // args from index 1+ are all constants, such as in TopkKudaf
-      final List<String> args = functionArgs
-          .stream()
-          .map(Expression::toString)
-          .collect(Collectors.toList());
+      final AggregateFunctionInitArguments aggregateFunctionInitArguments =
+          createAggregateFunctionInitArgs(udafIndex.getAsInt(), functionCall);
 
       return functionRegistry.getAggregateFunction(
           functionCall.getName().name(),
           argumentType,
-          AggregateFunctionInitArguments.ofFunctionArgs(
-              udafIndex.orElseThrow(
-                  () -> new KsqlException("Could not find column for expression: " + arg)),
-              args)
+          aggregateFunctionInitArguments
       );
     } catch (final Exception e) {
       throw new KsqlException("Failed to create aggregate function: " + functionCall, e);
     }
   }
+
+  public static AggregateFunctionInitArguments createAggregateFunctionInitArgs(
+      final int udafIndex,
+      final FunctionCall functionCall
+  ) {
+    // args from index > 0 are all literals
+    final List<Object> args = functionCall.getArguments()
+        .stream()
+        .skip(1)
+        .map(expr -> {
+          if (expr instanceof Literal) {
+            return (Literal)expr;
```

Comment:
I'm wondering if this is the same behavior as before 🤔 e.g. `NullLiteral` would now return `null` when previously it would return `"null"`. I'm guessing this is OK? That's the only difference I could find with a quick look - other expressions would probably be nonsensical if we did `toString` on them... commenting so maybe you'd have any thoughts, but I don't think it's a problem.

Reply:
I think the only udaf we have that takes init args is the topk/topkdistinct one which takes an in, so most probably we just didn't know how weird the init arg passing was before.